monai/bundle/scripts.py

```diff
@@ -32,14 +38,49 @@ def _update_default_args(args: Optional[Union[str, Dict]] = None, **kwargs) -> D
 
     # recursively update the default args with new args
     for k, v in kwargs.items():
-        args_[k] = _update_default_args(args_[k], **v) if isinstance(v, dict) and isinstance(args_.get(k), dict) else v
+        if ignore_none and v is None:
+            continue
+        if isinstance(v, dict) and isinstance(args_.get(k), dict):
+            args_[k] = _update_args(args_[k], ignore_none, **v)
+        else:
+            args_[k] = v
     return args_
 
 
+def _log_input_summary(tag: str, args: Dict):
+    logger.info(f"\n--- input summary of monai.bundle.scripts.{tag} ---")
+    for name, val in args.items():
+        logger.info(f"> {name}: {pprint.pformat(val)}")
+    logger.info("---\n\n")
+
+
+def _get_fake_spatial_shape(shape: Sequence[Union[str, int]], p: int = 1, n: int = 1, any: int = 1) -> Tuple[int]:
+    """
+    Get spatial shape for fake data according to the specified shape pattern.
+    It supports `int` number and `string` with formats like: "32", "32 * n", "32 ** p", "32 ** p *n".
+
+    Args:
+        shape: specified pattern for the spatial shape.
+        p: power factor to generate fake data shape if dim of expected shape is "x**p", default to 1.
+        p: multiply factor to generate fake data shape if dim of expected shape is "x*n", default to 1.
+        any: specified size to generate fake data shape if dim of expected shape is "*", default to 1.
+
+    """
+    ret = []
+    for i in shape:
+        if isinstance(i, int):
+            ret.append(i)
+        elif isinstance(i, str):
+            ret.append(any if i == "*" else eval(i, {"p": p, "n": n}))
```

Comment:
I was thinking that we would want to allow any arbitrary variables and not hard-code `m` and `p` as the only multiples and powers that we could have. It's gets very complicated what sort of expression we'd allow and then NP-hard to find values that satisfy a set of spatial size expressions, but I still think it's tractable to do it for our small amount of data. One thing we could use is the `ast` module to parse the expression and discover what variables are present, such as:

```python
import ast
def get_names(expr):
    tree=ast.parse(expr)
    names=[m.id for m in ast.walk(tree) if isinstance(m, ast.Name)]
    return names

get_names("2**p*m+1+2*v")  # returns ['v', 'm', 'p']
```

Reply:
Added the check in the latest commit.

Thanks for the suggestion.